### PR TITLE
Pass context when instanciating a serializer

### PR DESCRIPTION
**Pull request by ticosax**
Hi,
Thank you for this project, I'm very impressed.

Some of our serializers needs the request to render properly, this change aim to fix that.

The patch is inspired by DRF itself.
https://github.com/encode/django-rest-framework/blob/master/rest_framework/generics.py#L111

### CODE REVIEW

**Comment by axnsan12** [comment]
Hello and thank you!

Looks good and makes sense at first glance.

Could you provide some examples for this usage? Perhaps even some tests would be useful to make sure this keeps working in the future.

**Comment by ticosax** [comment]
sure thing. I will add the test.
the time being, here is the link explaining why it might comes handy.
http://www.django-rest-framework.org/api-guide/serializers/#including-extra-context

File: src/drf_yasg/inspectors.py [hunk 1]
```diff
@@ -114,7 +114,8 @@ def get_view_serializer(self):
         """
         if not hasattr(self.view, 'get_serializer'):
             return None
-        return self.view.get_serializer()
```
Comment (axnsan12) [inline src/drf_yasg/inspectors.py]: I don't follow, though. In [the link you sent](https://github.com/encode/django-rest-framework/blob/a81e60ff390b9b6788b2cac24a01ee4dd2dcffd2/rest_framework/generics.py#L111), you can see that `get_serializer` already adds the serializer context before instantiating `serializer_class`.

File: src/drf_yasg/inspectors.py [hunk 2]
```diff
@@ -114,7 +114,8 @@ def get_view_serializer(self):
...
+        kwargs = {'context': self.view.get_serializer_context()}
```
Comment (axnsan12) [inline src/drf_yasg/inspectors.py]: In fact, this line is exactly like the line you linked to, and it is in the same function that is called above.

**Comment by axnsan12** [comment]
I think the real problem you are encountering is not that context is not added to the serialzier, but that `context['request']` is `None`. Unfortunately, this is just the way it is, there is no request when the schema generator inspects views, because they are artificially instantiated.

What do you mean that your serializer "needs the request to render properly"? Could you provide an example of such a serializer?

File: src/drf_yasg/inspectors.py (hunk 1 above)
Comment (ticosax) [reply src/drf_yasg/inspectors.py]: damn, you are right!

**Comment by ticosax** [comment]
request is None when the generator is configured with `public=True`

https://github.com/axnsan12/drf-yasg/blob/master/src/drf_yasg/generators.py#L85

That's in fact enough to have the request.
Thank you.

**Comment by axnsan12** [comment]
Do note that this request will be, for all views, the request that arrived **against the schema view** (e.g. `GET /swagger.json`) and not a request that would normally hit your serializer, with query parameters, different HTTP methods or whatnot.

**Comment by ticosax** [comment]
Sure, as a side effect we have a different documentation rendered, depending if user is connected or not and so on.
Our permission policy is reflected on the documentation that way.

**Comment by axnsan12** [comment]
In 1.0.6, views should now always have the schema view request bound, regardless of `public=True`.

**Pull request closed without merging**